### ISSUE TITLE
SequenceVectors: learning rate decay over epochs

### ISSUE THREAD

**Issue by silvioOlivastri** [deeplearning4j/deeplearning4j#3278]
Hi,

trying to learn how SequenceVectors works I look a possible issue about learning rate decay over epochs. For example, take the `org.deeplearning4j.examples.nlp.paragraphvectors.ParagraphVectorsTextExample` class and insert `.epochs(2)` in `ParagraphVectors.Builder()`, the output was:

```
o.d.m.s.SequenceVectors - Starting learning process...
o.d.m.s.SequenceVectors - Epoch: [1]; Words vectorized so far: [654018];  Lines vectorized so far: [100003]; learningRate: [0.020529591097379254]
o.d.m.s.SequenceVectors - Epoch: [1]; Words vectorized so far: [1306480];  Lines vectorized so far: [200000]; learningRate: [0.0160695175468577]
o.d.m.s.SequenceVectors - Epoch: [1]; Words vectorized so far: [1959378];  Lines vectorized so far: [300000]; learningRate: [0.01160662091404979]
o.d.m.s.SequenceVectors - Epoch: [1]; Words vectorized so far: [2611349];  Lines vectorized so far: [400002]; learningRate: [0.007150245396256855]
o.d.m.s.SequenceVectors - Epoch: [1]; Words vectorized so far: [3171540];  Lines vectorized so far: [485810]; learningRate: [1.0E-4]
o.d.m.s.SequenceVectors - Epoch: [2]; Words vectorized so far: [653947];  Lines vectorized so far: [100004]; learningRate: [0.020530110599204343]
o.d.m.s.SequenceVectors - Epoch: [2]; Words vectorized so far: [1306586];  Lines vectorized so far: [200000]; learningRate: [0.016068875005126663]
o.d.m.s.SequenceVectors - Epoch: [2]; Words vectorized so far: [1959205];  Lines vectorized so far: [300000]; learningRate: [0.01160778979315625]
o.d.m.s.SequenceVectors - Epoch: [2]; Words vectorized so far: [2611128];  Lines vectorized so far: [400001]; learningRate: [0.007151749217329487]
o.d.m.s.SequenceVectors - Epoch: [2]; Words vectorized so far: [3171540];  Lines vectorized so far: [485810]; learningRate: [1.0E-4]
```

This leaves me thinking that in each epoch the words at the end of data set is less train than the initial words.

This kind of "feature" it's desired or is an issue?

IMHO learning rate decay should be doing over epochs but in the same epoch it should not change, for example (handwritten):

```
o.d.m.s.SequenceVectors - Starting learning process...
o.d.m.s.SequenceVectors - Epoch: [1]; Words vectorized so far: [654018];  Lines vectorized so far: [100003]; learningRate: [0.025]
o.d.m.s.SequenceVectors - Epoch: [1]; Words vectorized so far: [1959378];  Lines vectorized so far: [300000]; learningRate: [0.025]
o.d.m.s.SequenceVectors - Epoch: [1]; Words vectorized so far: [3171540];  Lines vectorized so far: [485810]; learningRate: [0.025]
o.d.m.s.SequenceVectors - Epoch: [2]; Words vectorized so far: [654018];  Lines vectorized so far: [100003]; learningRate: [0.022]
o.d.m.s.SequenceVectors - Epoch: [2]; Words vectorized so far: [1959378];  Lines vectorized so far: [300000]; learningRate: [0.022]
o.d.m.s.SequenceVectors - Epoch: [2]; Words vectorized so far: [3171540];  Lines vectorized so far: [485810]; learningRate: [0.022]
o.d.m.s.SequenceVectors - Epoch: [3]; Words vectorized so far: [654018];  Lines vectorized so far: [100003]; learningRate: [0.019]
o.d.m.s.SequenceVectors - Epoch: [3]; Words vectorized so far: [1959378];  Lines vectorized so far: [300000]; learningRate: [0.019]
o.d.m.s.SequenceVectors - Epoch: [3]; Words vectorized so far: [3171540];  Lines vectorized so far: [485810]; learningRate: [0.019]
..........
```

Arrive at `minLearningRate()` or not depends on decay function and the number of the epochs used.

DL4J version: 0.8

**Comment by raver119** [comment]
Agreed and applied requested changes: https://github.com/deeplearning4j/deeplearning4j/pull/3642
